lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -111,7 +113,51 @@ export default function FilePickerApp({ onSubmit }) {
    * render.
    */
   const [shouldSubmit, setShouldSubmit] = useState(false);
-  const submit = useCallback(() => setShouldSubmit(true), []);
+
+  const [deeplinkingFields, setDeeplinkingFields] = useState(
+    /** @type {Record<string,string>} */ null
+  );
+
+  const submit = useCallback(
+    async (/** @type {Content} */ content) => {
+      async function getDeeplinkingFields() {
+        const data = {
+          ...deeplinkingAPI.data,
+          content,
+          extra_params: {
+            groupSet: groupConfig.useGroupSet ? groupConfig.groupSet : null,
+          },
+        };
+        setDeeplinkingFields(
+          await apiCall({
+            authToken: deeplinkingAPI.authToken,
+            path: deeplinkingAPI.path,
+            data,
+          })
+        );
+      }
+      // When deeplinkingAPI is present we want to call the backend to return the form
+      // fields we'll forward to the LMS to complete the DeepLinking request
+      if (content && deeplinkingAPI && !deeplinkingFields) {
```

Comment:
`deeplinkingAPI` is a mandatory field according to the updated `FilePickerConfig` type. Is that correct? If so it doesn't need to be checked for here.

Reply:
It won't be present for LTI1.1 launches on the first PR.